Document JRuby incompatibility due to JRuby Open3 bug

Bug in `Open3` mean this doesn't work in JRuby 1.7, due to be fixed in 1.7.18

https://github.com/jruby/jruby/issues/1290
https://github.com/jruby/jruby/issues/2298

I am running jruby 1.7.19. I get the following warning: `jruby-1.7.19/lib/ruby/1.9/open3.rb:74 warning: popen3 does not support spawn options in JRuby 1.7.`

Browserify runs, but the paths for requiring are messed up. I have to require all of the files as if I were in the root directory of the app, so like: `require('./app/assets/javascripts/...')`.

If I use the normal paths for requiring, I get this error:

```
Browserify: /Users/joseph.adams/Documents/RubymineProjects/patient_summary/node_modules/.bin/browserifyinc --transform reactify --extension=".jsx" --list --cachefile="/Users/joseph.adams/Documents/RubymineProjects/patient_summary/tmp/cache/browserify-rails/browserifyinc-cache.json" -o "/Users/joseph.adams/Documents/RubymineProjects/patient_summary/tmp/cache/browserify-rails/output20150323-71049-13h4217" -
/Users/joseph.adams/.rvm/rubies/jruby-1.7.19/lib/ruby/1.9/open3.rb:74 warning: popen3 does not support spawn options in JRuby 1.7
Browserify: /Users/joseph.adams/Documents/RubymineProjects/patient_summary/node_modules/.bin/browserifyinc --transform reactify --extension=".jsx" --list --cachefile="/Users/joseph.adams/Documents/RubymineProjects/patient_summary/tmp/cache/browserify-rails/browserifyinc-cache.json" -o "/Users/joseph.adams/Documents/RubymineProjects/patient_summary/tmp/cache/browserify-rails/output20150323-71049-13h4217" -
-------------------------------------------------------------------
EXCEPTION - Error Tracking ID: d5ce41f0-fa25-4867-b3a3-61d28948ae68
class: BrowserifyRails::BrowserifyError
message: Error while running `/Users/joseph.adams/Documents/RubymineProjects/patient_summary/node_modules/.bin/browserifyinc --transform reactify --extension=".jsx" --list --cachefile="/Users/joseph.adams/Documents/RubymineProjects/patient_summary/tmp/cache/browserify-rails/browserifyinc-cache.json" -o "/Users/joseph.adams/Documents/RubymineProjects/patient_summary/tmp/cache/browserify-rails/output20150323-71049-13h4217" -`:

events.js:85
      throw er; // Unhandled 'error' event
            ^
Error: Cannot find module './greeter.js' from '/Users/joseph.adams/Documents/RubymineProjects/patient_summary'
    at /Users/joseph.adams/Documents/RubymineProjects/patient_summary/node_modules/browserify-incremental/node_modules/browserify/node_modules/browser-resolve/node_modules/resolve/lib/async.js:55:21
    at load (/Users/joseph.adams/Documents/RubymineProjects/patient_summary/node_modules/browserify-incremental/node_modules/browserify/node_modules/browser-resolve/node_modules/resolve/lib/async.js:69:43)
    at onex (/Users/joseph.adams/Documents/RubymineProjects/patient_summary/node_modules/browserify-incremental/node_modules/browserify/node_modules/browser-resolve/node_modules/resolve/lib/async.js:92:31)
    at /Users/joseph.adams/Documents/RubymineProjects/patient_summary/node_modules/browserify-incremental/node_modules/browserify/node_modules/browser-resolve/node_modules/resolve/lib/async.js:22:47
    at Object.oncomplete (fs.js:93:15)

  (in /Users/joseph.adams/Documents/RubymineProjects/patient_summary/app/assets/javascripts/modern_spa.js)

stacktrace:
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/browserify-rails-0.8.1/lib/browserify-rails/browserify_processor.rb:168:in `run_browserify'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/browserify-rails-0.8.1/lib/browserify-rails/browserify_processor.rb:104:in `dependencies'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/browserify-rails-0.8.1/lib/browserify-rails/browserify_processor.rb:81:in `commonjs_module?'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/browserify-rails-0.8.1/lib/browserify-rails/browserify_processor.rb:61:in `should_browserify?'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/browserify-rails-0.8.1/lib/browserify-rails/browserify_processor.rb:14:in `evaluate'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/tilt-1.4.1/lib/tilt/template.rb:103:in `render'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-2.12.3/lib/sprockets/context.rb:197:in `evaluate'
org/jruby/RubyArray.java:1613:in `each'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-2.12.3/lib/sprockets/context.rb:194:in `evaluate'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-2.12.3/lib/sprockets/processed_asset.rb:12:in `initialize'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-2.12.3/lib/sprockets/base.rb:374:in `build_asset'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-2.12.3/lib/sprockets/base.rb:395:in `circular_call_protection'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-2.12.3/lib/sprockets/base.rb:373:in `build_asset'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-2.12.3/lib/sprockets/index.rb:94:in `build_asset'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-2.12.3/lib/sprockets/caching.rb:55:in `cache_asset'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-2.12.3/lib/sprockets/index.rb:93:in `build_asset'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-2.12.3/lib/sprockets/base.rb:287:in `find_asset'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-2.12.3/lib/sprockets/index.rb:60:in `find_asset'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-2.12.3/lib/sprockets/bundled_asset.rb:16:in `initialize'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-2.12.3/lib/sprockets/base.rb:377:in `build_asset'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-2.12.3/lib/sprockets/index.rb:94:in `build_asset'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-2.12.3/lib/sprockets/caching.rb:55:in `cache_asset'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-2.12.3/lib/sprockets/index.rb:93:in `build_asset'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-2.12.3/lib/sprockets/base.rb:287:in `find_asset'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-2.12.3/lib/sprockets/index.rb:60:in `find_asset'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-2.12.3/lib/sprockets/environment.rb:75:in `find_asset'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-2.12.3/lib/sprockets/base.rb:295:in `[]'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-rails-2.2.4/lib/sprockets/rails/helper.rb:230:in `lookup_asset_for_path'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-rails-2.2.4/lib/sprockets/rails/helper.rb:190:in `check_errors_for'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-rails-2.2.4/lib/sprockets/rails/helper.rb:89:in `asset_path'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionview-4.1.1/lib/action_view/helpers/asset_url_helper.rb:234:in `javascript_path'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionview-4.1.1/lib/action_view/helpers/asset_tag_helper.rb:60:in `javascript_include_tag'
org/jruby/RubyArray.java:2412:in `map'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionview-4.1.1/lib/action_view/helpers/asset_tag_helper.rb:59:in `javascript_include_tag'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/sprockets-rails-2.2.4/lib/sprockets/rails/helper.rb:148:in `javascript_include_tag'
/Users/joseph.adams/Documents/RubymineProjects/patient_summary/app/views/layouts/application.html.erb:44:in `_app_views_layouts_application_html_erb___1584920363_6594'
org/jruby/RubyProc.java:271:in `call'
/Users/joseph.adams/Documents/RubymineProjects/patient_summary/app/helpers/javascript_helper.rb:20:in `conditional_html_for_modern_browsers_only'
/Users/joseph.adams/Documents/RubymineProjects/patient_summary/app/views/layouts/application.html.erb:42:in `_app_views_layouts_application_html_erb___1584920363_6594'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionview-4.1.1/lib/action_view/template.rb:145:in `render'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/activesupport-4.1.1/lib/active_support/notifications.rb:161:in `instrument'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionview-4.1.1/lib/action_view/template.rb:339:in `instrument'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionview-4.1.1/lib/action_view/template.rb:143:in `render'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/rack-mini-profiler-0.9.3/lib/mini_profiler/profiling_methods.rb:106:in `render_with_mini_profiler'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionview-4.1.1/lib/action_view/renderer/template_renderer.rb:67:in `render_with_layout'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionview-4.1.1/lib/action_view/renderer/template_renderer.rb:53:in `render_template'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionview-4.1.1/lib/action_view/renderer/template_renderer.rb:17:in `render'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionview-4.1.1/lib/action_view/renderer/renderer.rb:42:in `render_template'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionview-4.1.1/lib/action_view/renderer/renderer.rb:23:in `render'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionview-4.1.1/lib/action_view/rendering.rb:99:in `_render_template'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_controller/metal/streaming.rb:217:in `_render_template'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionview-4.1.1/lib/action_view/rendering.rb:82:in `render_to_body'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_controller/metal/rendering.rb:32:in `render_to_body'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_controller/metal/renderers.rb:32:in `render_to_body'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/abstract_controller/rendering.rb:25:in `render'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_controller/metal/rendering.rb:16:in `render'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_controller/metal/instrumentation.rb:41:in `render'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/activesupport-4.1.1/lib/active_support/core_ext/benchmark.rb:12:in `ms'
/Users/joseph.adams/.rvm/rubies/jruby-1.7.19/lib/ruby/1.9/benchmark.rb:295:in `realtime'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/activesupport-4.1.1/lib/active_support/core_ext/benchmark.rb:12:in `ms'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_controller/metal/instrumentation.rb:41:in `render'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_controller/metal/instrumentation.rb:84:in `cleanup_view_runtime'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/activerecord-4.1.1/lib/active_record/railties/controller_runtime.rb:25:in `cleanup_view_runtime'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_controller/metal/instrumentation.rb:40:in `render'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/wicked_pdf-0.11.0/lib/wicked_pdf/pdf_helper.rb:23:in `render_with_wicked_pdf'
/Users/joseph.adams/Documents/RubymineProjects/patient_summary/app/controllers/application_controller.rb:37:in `break_the_glass_warn'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_controller/metal/implicit_render.rb:4:in `send_action'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/abstract_controller/base.rb:189:in `process_action'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_controller/metal/rendering.rb:10:in `process_action'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/abstract_controller/callbacks.rb:20:in `process_action'
org/jruby/RubyProc.java:271:in `call'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/activesupport-4.1.1/lib/active_support/callbacks.rb:113:in `call'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/activesupport-4.1.1/lib/active_support/callbacks.rb:166:in `halting'
org/jruby/RubyProc.java:271:in `call'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/activesupport-4.1.1/lib/active_support/callbacks.rb:149:in `halting_and_conditional'
org/jruby/RubyProc.java:271:in `call'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/activesupport-4.1.1/lib/active_support/callbacks.rb:166:in `halting'
org/jruby/RubyProc.java:271:in `call'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/activesupport-4.1.1/lib/active_support/callbacks.rb:229:in `halting'
org/jruby/RubyProc.java:271:in `call'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/activesupport-4.1.1/lib/active_support/callbacks.rb:229:in `halting'
org/jruby/RubyProc.java:271:in `call'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/activesupport-4.1.1/lib/active_support/callbacks.rb:166:in `halting'
org/jruby/RubyProc.java:271:in `call'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/activesupport-4.1.1/lib/active_support/callbacks.rb:86:in `run_callbacks'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/abstract_controller/callbacks.rb:19:in `process_action'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_controller/metal/rescue.rb:29:in `process_action'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_controller/metal/instrumentation.rb:31:in `process_action'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/activesupport-4.1.1/lib/active_support/notifications.rb:159:in `instrument'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/activesupport-4.1.1/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/activesupport-4.1.1/lib/active_support/notifications.rb:159:in `instrument'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_controller/metal/instrumentation.rb:30:in `process_action'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_controller/metal/params_wrapper.rb:250:in `process_action'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/activerecord-4.1.1/lib/active_record/railties/controller_runtime.rb:18:in `process_action'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/abstract_controller/base.rb:136:in `process'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionview-4.1.1/lib/action_view/rendering.rb:30:in `process'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/rack-mini-profiler-0.9.3/lib/mini_profiler/profiling_methods.rb:106:in `process_with_mini_profiler'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_controller/metal.rb:195:in `dispatch'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_controller/metal/rack_delegation.rb:13:in `dispatch'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_controller/metal.rb:231:in `action'
org/jruby/RubyProc.java:271:in `call'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_dispatch/routing/route_set.rb:80:in `dispatch'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_dispatch/routing/route_set.rb:48:in `call'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_dispatch/journey/router.rb:71:in `call'
org/jruby/RubyArray.java:1613:in `each'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_dispatch/journey/router.rb:59:in `call'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_dispatch/routing/route_set.rb:676:in `call'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/rack-1.5.2/lib/rack/content_length.rb:14:in `call'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/rack-1.5.2/lib/rack/etag.rb:23:in `call'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/rack-1.5.2/lib/rack/conditionalget.rb:25:in `call'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/rack-1.5.2/lib/rack/head.rb:11:in `call'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_dispatch/middleware/params_parser.rb:27:in `call'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_dispatch/middleware/flash.rb:254:in `call'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/rack-1.5.2/lib/rack/session/abstract/id.rb:225:in `context'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/rack-1.5.2/lib/rack/session/abstract/id.rb:220:in `call'
/Users/joseph.adams/.rvm/gems/jruby-1.7.19/gems/actionpack-4.1.1/lib/action_dispatch/middleware/cookies.rb:560:in `call'
```

Is there any way I can fix this issue, or is it just an underlying issue with jruby?

--- a/lib/browserify-rails/browserify_processor.rb
+++ b/lib/browserify-rails/browserify_processor.rb
@@ -171,7 +171,15 @@ module BrowserifyRails
       base_directory = File.dirname(file)
 
       Logger::log "Browserify: #{command}"
-      stdout, stderr, status = Open3.capture3(env, command, stdin_data: data, chdir: base_directory)
+
+      # If we are on JRuby 1.x, capture3 does not support chdir option
+      stdout, stderr, status = if RUBY_PLATFORM == "java" && JRUBY_VERSION =~ /^1/
+        Dir.chdir(base_directory) {
+          Open3.capture3(env, command, stdin_data: data)
+        }
+      else
+        Open3.capture3(env, command, stdin_data: data, chdir: base_directory)
+      end
 
       if !status.success?
         raise BrowserifyRails::BrowserifyError.new("Error while running `#{command}`:\n\n#{stderr}")

